presto-kinesis/src/main/java/io/prestosql/plugin/kinesis/s3config/S3TableConfigClient.java

```diff
@@ -0,0 +1,243 @@
+            ObjectListing result;
+
+            do {
+                result = s3client.listObjects(req);
+
+                returnList.addAll(result.getObjectSummaries());
+                req.setMarker(result.getNextMarker());
+            }
+            while (result.isTruncated());
+
+            log.info("Completed getting S3 object listing.");
+        }
+        catch (AmazonServiceException amazonServiceException) {
+            log.error(amazonServiceException.toString());
+        }
+        catch (AmazonClientException amazonClientException) {
+            log.error(amazonClientException.toString());
+        }
+
+        return returnList;
+    }
+
+    /**
+     * Connect to S3 directory to look for new or updated table definitions and then
+     * update the map.
+     */
+    protected void updateTablesFromS3()
+    {
+        long now = System.currentTimeMillis();
+
+        List<S3ObjectSummary> objectList = this.getObjectSummaries();
+        AmazonS3Client s3client = this.clientManager.getS3Client();
+        //AmazonS3URI directoryURI = new AmazonS3URI(this.bucketUrl.get());
+
+        for (S3ObjectSummary objInfo : objectList) {
+            if (!this.internalMap.containsKey(objInfo.getKey()) || objInfo.getLastModified().getTime() >= this.lastCheck) {
+                // New or updated file, so we must read from AWS
+                try {
+                    if (objInfo.getKey().endsWith("/")) {
+                        continue;
+                    }
+
+                    log.info("Getting : %s - %s", objInfo.getBucketName(), objInfo.getKey());
+                    S3Object object = s3client.getObject(
+                            new GetObjectRequest(objInfo.getBucketName(), objInfo.getKey()));
+
+                    StringBuilder resultStr = new StringBuilder("");
+                    try (BufferedReader reader = new BufferedReader(new InputStreamReader(object.getObjectContent()))) {
+                        boolean hasMore = true;
+                        while (hasMore) {
+                            String line = reader.readLine();
+                            if (line != null) {
+                                resultStr.append(line);
+                            }
+                            else {
+                                hasMore = false;
+                            }
+                        }
+
+                        KinesisStreamDescription table = streamDescriptionCodec.fromJson(resultStr.toString());
+
+                        internalMap.put(objInfo.getKey(), table);
+                        log.info("Put table description into the map from %s", objInfo.getKey());
+                    }
+                    catch (IOException iox) {
+                        log.error("Problem reading input stream from object.", iox);
+                    }
+                }
+                catch (AmazonServiceException ase) {
+                    log.error(ase, "Request to S3 was rejected");
```

Comment:
This should probably be a hard failure. Otherwise, a glitch in talking to AWS will cause tables to go missing or other subtle issues.

Reply:
removed the catches